Source/WebCore/style/StyleResolver.cpp

```diff
@@ -726,24 +726,59 @@ std::optional<DynamicMediaQueryEvaluationChanges> Resolver::evaluateDynamicMedia
 }
 
 
-void Resolver::setViewTransitionGroupStyles(const AtomString& name, Ref<MutableStyleProperties> properties)
+static CSSSelectorList viewTransitionSelector(CSSSelector::PseudoElement element, const AtomString& name)
 {
-    if (!m_document)
-        return;
+    MutableCSSSelectorList selectorList;
 
-    auto* viewTransitionsStyle = m_ruleSets.dynamicViewTransitionsStyle();
-    RuleSetBuilder builder(*viewTransitionsStyle, mediaQueryEvaluator(), this);
-    CSSSelectorParserContext context(*m_document.get());
+    auto rootSelector = makeUnique<MutableCSSSelector>();
+    rootSelector->setMatch(CSSSelector::Match::PseudoClass);
+    rootSelector->setPseudoClass(CSSSelector::PseudoClass::Root);
+    selectorList.append(WTFMove(rootSelector));
+
+    auto groupSelector = makeUnique<MutableCSSSelector>();
+    groupSelector->setMatch(CSSSelector::Match::PseudoElement);
+    groupSelector->setPseudoElement(element);
+
+    AtomString selectorName;
+    switch (element) {
+    case CSSSelector::PseudoElement::ViewTransition:
+        selectorName = "view-transition"_s;
+        break;
+    case CSSSelector::PseudoElement::ViewTransitionGroup:
+        selectorName = "view-transition-group"_s;
+        break;
+    case CSSSelector::PseudoElement::ViewTransitionImagePair:
+        selectorName = "view-transition-image-pair"_s;
+            break;
+    case CSSSelector::PseudoElement::ViewTransitionNew:
+        selectorName = "view-transition-new"_s;
+        break;
+    case CSSSelector::PseudoElement::ViewTransitionOld:
+        selectorName = "view-transition-old"_s;
+        break;
+    default:
+        ASSERT_NOT_REACHED();
+        break;
+    }
 
-    MutableCSSSelectorList selectorList;
-    selectorList.append(MutableCSSSelector::parsePseudoClassSelector("root"_s, context));
-    auto groupSelector = MutableCSSSelector::parsePseudoElementSelector("view-transition-group"_s, context);
+    groupSelector->setValue(selectorName);
     groupSelector->setArgumentList({ { name } });
```

Comment:
You don't want to set the argument list when the selector name is `view-transition`